### DISCUSSION TITLE
Add node and edge drop and Connection Radius

### DISCUSSION THREAD

**Discussion by Gatts011** [xyflow/xyflow#3497] [General]
Hi There, noticed on this example [AddNodeOnedgeDrop](https://reactflow.dev/docs/examples/nodes/add-node-on-edge-drop/) if the handle is a small circle, the edge snaps within 20 pixels but finds pane under mouse and adds a node when you'd be expecting a connection. 
Is there a way to modify the example to see the connection radius under mouse instead of pane?
OR
If connection is true dont add node before connectEnd?

Thanks
G

**Comment by bcakmakoglu** [comment]
I think the simplest solution would be to just use a flag indicating whether a connection was just made or not.

```ts
const wasConnected = useRef(false);

const onConnect = useCallback((params) => {
  setEdges((eds) => addEdge(params, eds));
  
  // set the flag to true
  wasConnected.current = true;
}, []);

const onConnectEnd = useCallback(
  (event) => {
    const targetIsPane = event.target.classList.contains("react-flow__pane");

	// if a connection was just made, we don't want to add a node
    if (targetIsPane && !wasConnected.current) {
      // ...
    }

	// reset the flag
    wasConnected.current = false;
  },
  [project]
);
```

[Here's an example.](https://codesandbox.io/s/ancient-worker-v7qmmr?file=/App.js:1100-2006)

**Reply by Gatts011** [reply]
Works perfectly, thank you @bcakmakoglu!!!
